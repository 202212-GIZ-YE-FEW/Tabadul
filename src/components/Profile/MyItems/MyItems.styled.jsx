import styled from "@emotion/styled";
import Image from "next/image";

export const Header = styled.h1`
    text-align: center;
    font-style: normal;
    font-weight: 700;
    font-size: 56px;
    line-height: 84px;
    color: #3c4347;
    margin-bottom: 90px;
    @media screen and (max-width: 768px) {
        font-size: 36px;
    }
`;

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 35px;
`;

export const ItemCard = styled.div`
    display: flex;
    flex-direction: row;
    background: #f5f5f5;
    border-radius: 46px;
    width: 1458px;
    height: 104.63px;
    max-width: 100%;
    margin-bottom: 40px;
    @media screen and (max-width: 768px) {
        height: 100%;
        align-items: center;
        padding: 3px 14px 0px 2px;
        justify-content: space-between;
    }
`;

export const ItemImage = styled(Image)`
    height: 100px;
    width: 100px;
    border-radius: 50%;
    margin-right: 50px;
    padding: 5px;
    @media screen and (max-width: 768px) {
        margin-right: 20px;
        height: 73px;
        width: 48px;
    }
`;

export const Text = styled.span`
    font-style: normal;
    font-weight: 400;
    font-size: 24px;
    line-height: 20px;
    display: flex;
    align-items: center;
    @media screen and (max-width: 768px) {
        font-size: 20px;
    }
`;

export const Icon = styled(Image)`
    /* display: flex;
    align-items: center;
    justify-content: center; */
    margin: 35px;
    cursor: pointer;
    @media screen and (max-width: 768px) {
        margin: 0px;
        width: 28px;
        height: 29px;
        margin-left: 20px;
    }
`;
export const NoData = styled(Image)`
    margin: 0 auto;
    display: flex;
    place-content: center;
    max-width: 100%;
    @media screen and (max-width: 786px) {
        width: 70%;
    }
`;
