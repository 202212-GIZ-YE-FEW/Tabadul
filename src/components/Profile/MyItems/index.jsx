import { collection, deleteDoc, doc, getDocs } from "firebase/firestore";
import Link from "next/link";
import React, { useEffect, useState } from "react";
import swal from "sweetalert";

import { auth, db } from "@/utils/firebase";

import {
    Container,
    Header,
    Icon,
    ItemCard,
    ItemImage,
    Text,
} from "./MyItems.styled";
import deleteIcon from "../../../../public/images/deleteIcon.svg";
import penIcon from "../../../../public/images/itemsPen.svg";

const MyItems = () => {
    const [relateditems, setrelateditems] = useState();
    const itemsCollRef = collection(db, "items");
    useEffect(() => {
        const getallitems = async () => {
            try {
                const itemsData = await getDocs(itemsCollRef);
                const itemsDocs = itemsData.docs
                    .map((loc) => {
                        return { ...loc.data(), id: loc.id };
                    })
                    .filter((ele) => {
                        return ele.userId === auth.currentUser.uid;
                    });
                setrelateditems(itemsDocs[0]);
            } catch (err) {
                console.error(err);
            }
        };
        getallitems();
    }, []);
    const deleteProduct = async (id) => {
        const productDoc = doc(db, "items", id);
        await deleteDoc(productDoc);
        try {
            swal({
                title: "Product Detected!",
                text: "Product has been Detected",
                icon: "success",
                button: "Cancel",
            });
        } catch (error) {
            console.log(error);
        }
    };
    return (
        <>
            <Header>My Items</Header>
            {relateditems && (
                <Container>
                    <ItemCard>
                        <ItemImage
                            src={relateditems?.image[0].url}
                            alt='RealtedIcon'
                            width={100}
                            height={100}
                        />
                        <Text>{relateditems?.title}</Text>
                        <Link
                            href='/Updateproduct'
                            style={{ marginLeft: "auto" }}
                        >
                            <Icon src={penIcon} alt='pen Icon' />
                        </Link>
                        <Icon
                            src={deleteIcon}
                            alt='Delete Icon'
                            onClick={() => deleteProduct(relateditems.id)}
                        />
                    </ItemCard>
                </Container>
            )}
        </>
    );
};

export default MyItems;
