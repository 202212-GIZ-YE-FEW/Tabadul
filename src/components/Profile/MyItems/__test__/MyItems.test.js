import renderer from "react-test-renderer";

import MyItems from "..";

it("renders correctly", () => {
    const tree = renderer.create(<MyItems />).toJSON();
    expect(tree).toMatchSnapshot();
});
