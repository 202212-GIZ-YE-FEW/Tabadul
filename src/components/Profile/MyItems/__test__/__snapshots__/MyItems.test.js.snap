// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1
    className="css-127jica"
  >
    My Items
  </h1>,
  <img
    className="css-1ia6pzt"
    data-nimg="1"
    decoding="async"
    height={40}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style={
      Object {
        "color": "transparent",
      }
    }
    width={40}
  />,
]
`;
